### ISSUE TITLE
Enhancement: Add the `NOT IN (SELECT *)` syntax

### ISSUE THREAD

**Issue by barrywhart** [alanmcruickshank/sqlfluff#71]
Query:
```
SELECT
    user_id AS non_GDPR_users
FROM
    p.d.t AS query
WHERE user_id NOT IN (SELECT user_id FROM p.d.t2)
```

Output:
```
$ sqlfluff lint test.sql
== [test.sql] FAIL
L:   2 | P:  16 | L014 | Inconsistent capitalisation of unquoted identifiers.
L:   5 | P:  15 | ???? | Found unparsable segment @L005P015: 'NOT IN (SELECT user_...'
```

**Comment by alanmcruickshank** [comment]
This is less about the nested query operator, but about the implementation of the `NOT IN` syntax. I'll add it to the next release.

**Comment by alanmcruickshank** [comment]
This was closed as part of #83
